### ISSUE TITLE
False positive with "New value type is not in annotated types"?

### ISSUE THREAD

**Issue by jyggen** [kalessil/phpinspectionsea#756]
Consider the following code utilizing PSR-7 interfaces:
```php
<?php
use Psr\Http\Message\RequestInterface;

function withAddedHeader(RequestInterface $request, string $name, string $value): RequestInterface
{
    $request = $request->withAddedHeader($name, $value);

    return $request;
}
```

The following inspection is triggered on `$request->withAddedHeader()`:
```
The new value type (\Psr\Http\Message\MessageInterface) is not in annotated types.
```

`RequestInterface` extends `MessageInterface`, and `MessageInterface::withAddedHeader()` is documented to return `static`, so shouldn't `$request` still be considered `RequestInterface`?

**Comment by jyggen** [comment]
Now I get a similar error on `$request = $modifiedRequest;`, where `$modifiedRequest` is reported to be `static`:
```
New value type (static) is not in annotated types.
```

**Comment by kalessil** [comment]
Good, then it should be same as  same case as #746
